taichi/codegen/spirv/spirv_ir_builder.cpp

```diff
@@ -1181,8 +1181,8 @@ void IRBuilder::store_variable(Value pointer, Value value) {
 
 void IRBuilder::register_value(std::string name, Value value) {
   auto it = value_name_tbl_.find(name);
-  if (it != value_name_tbl_.end()) {
-    TI_ERROR("{} is existed.", name);
+  if (it != value_name_tbl_.end() && it->second.flag != ValueKind::kConstant) {
```

Comment:
OOC: does this mean we'll have two constant values with the same name? actually would you mind pasting a snippet of IR before and after this PR? Thanks!!

Reply:
> OOC: does this mean we'll have two constant values with the same name? actually would you mind pasting a snippet of IR before and after this PR? Thanks!! 

Well actually it's not quite a problem. OpName is technically a non-semantic op in SPIR-V so duplicated names do no harm. Duplicated constant declarations will be optimized out in a later processing by SPIR-Tools.